src/main/java/com/openshift/jenkins/plugins/pipeline/Auth.java

```diff
@@ -148,9 +157,47 @@ public boolean allowCertificate(final X509Certificate[] certificateChain) {
 
 	@Override
 	public boolean allowHostname(String hostname, SSLSession sslSession) {
-		//mimic SSLCertificateCallback implementation from jbosstools-openshift repo - we can noop
-		//also lines up with what was observed in k8s jennkins plugin
-		return true;
+        List<String> lookFor = new ArrayList<String>();
+        lookFor.add(hostname);
+        String hosts = env.get(SSL_PROXY_HOSTS);
+        if (LOGGER.isLoggable(Level.FINE))
+            LOGGER.log(Level.FINE, "allowHostname: ssl host override: " + hosts);
+        if (hosts != null) {
+            String[] hostNames = hosts.split(" ");
+            for (String host : hostNames) {
+                lookFor.add(host);
+            }
+        }
+        
+	    try {
+            for (String h : lookFor) {
+                // the hostname passed here is the hostname we use to initiate the ssl flow;
+                // note the javadoc for SSLSession.getPeerHost() says "This value is not authenticated and should not be relied upon";
+                // note SSLSession.getPeerPrincipal.getName() based on testing returns a CN=... subject name not useful to compare with;
+                // so we see if our hostname is in the list of any of ssl session's cert's subject names ... if so, we bless this
+                Certificate[] certs = sslSession.getPeerCertificates();
+                for (Certificate cert : certs) {
+                    if (cert instanceof X509Certificate) {
+                        X509Certificate x509cert = (X509Certificate)cert;
+                        Collection<List<?>> subjNames = x509cert.getSubjectAlternativeNames();
+                        if (subjNames != null) {
+                            for (List<?> list : subjNames) {
```

Comment:
this list isn't what you think it is... it contains two items, the first of which is an Integer telling you what type of subjectAlternativeName this is. You should only check DNS (type 2) and IP Address (type 7). The second element is the data you should compare the host against.